### ISSUE TITLE
无法重新定义用户文件夹

### ISSUE THREAD

**Issue by GoogleCodeExporter** [lotem/rimeime#270]
```
請寫下能夠重現此狀況的操作步驟：
1.重新定义用户文件夹至自己的一个文件夹
2.删除原本Roaming/Rime文件夹
3.无法调出中文字体输入。

所得的結果是這個樣：无法调出中文字体
預期結果該是那個樣：可以自己配置用户文件夹

請提供以下信息，幫我們定位問題。

1. 輸入法的款式：小狼毫
   輸入法版本號：0.9.15

2. 操作系統類型：Windows
   操作系統版本：win7  32bit

3. 輸入方案名稱：

還想再說幾句：弄了好几次，我把用户文件夹换回默认的路��
�就不会出现问题。
                          但是由于系统经常重装，所以放在C盘的roaming下不太放心，想单独导     出来以便备份。



```

Original issue reported on code.google.com by `shensh...@gmail.com` on 25 Sep 2012 at 11:36

**Comment by GoogleCodeExporter** [comment]
```
设置好之后，有没有退出【小狼毫算法服务】？你可以查看��
�志有没有从新位置加载。
%TEMP%\rime.weasel.*.INFO
```

Original comment by `chen....@gmail.com` on 25 Sep 2012 at 11:40

**Comment by GoogleCodeExporter** [comment]
```
Log file created at: 2012/09/25 19:51:32
Running on machine: WIN-8PDHNA3BDRA
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
I0925 19:51:32.171793  3192 RimeWithWeasel.cpp:42] Initializing la rime.
I0925 19:51:32.175794  3192 common.cc:49] registering built-in components
I0925 19:51:32.176795  3192 registry.cc:18] registering component: config
I0925 19:51:32.176795  3192 registry.cc:18] registering component: schema_config
I0925 19:51:32.176795  3192 registry.cc:18] registering component: dictionary
I0925 19:51:32.176795  3192 registry.cc:18] registering component: 
reverse_lookup_dictionary
I0925 19:51:32.176795  3192 registry.cc:18] registering component: 
user_dictionary
I0925 19:51:32.176795  3192 registry.cc:18] registering component: 
ascii_composer
I0925 19:51:32.176795  3192 registry.cc:18] registering component: 
chord_composer
I0925 19:51:32.176795  3192 registry.cc:18] registering component: 
express_editor
I0925 19:51:32.177794  3192 registry.cc:18] registering component: 
fluency_editor
I0925 19:51:32.177794  3192 registry.cc:18] registering component: key_binder
I0925 19:51:32.177794  3192 registry.cc:18] registering component: navigator
I0925 19:51:32.177794  3192 registry.cc:18] registering component: punctuator
I0925 19:51:32.177794  3192 registry.cc:18] registering component: recognizer
I0925 19:51:32.177794  3192 registry.cc:18] registering component: selector
I0925 19:51:32.177794  3192 registry.cc:18] registering component: speller
I0925 19:51:32.177794  3192 registry.cc:18] registering component: abc_segmentor
I0925 19:51:32.177794  3192 registry.cc:18] registering component: 
ascii_segmentor
I0925 19:51:32.177794  3192 registry.cc:18] registering component: matcher
I0925 19:51:32.177794  3192 registry.cc:18] registering component: 
punct_segmentor
I0925 19:51:32.177794  3192 registry.cc:18] registering component: 
fallback_segmentor
I0925 19:51:32.177794  3192 registry.cc:18] registering component: 
echo_translator
I0925 19:51:32.177794  3192 registry.cc:18] registering component: 
punct_translator
I0925 19:51:32.177794  3192 registry.cc:18] registering component: 
table_translator
I0925 19:51:32.177794  3192 registry.cc:18] registering component: 
r10n_translator
I0925 19:51:32.177794  3192 registry.cc:18] registering component: 
reverse_lookup_translator
I0925 19:51:32.177794  3192 registry.cc:18] registering component: simplifier
I0925 19:51:32.177794  3192 registry.cc:18] registering component: uniquifier
I0925 19:51:32.555815  3192 deployment_tasks.cc:31] updating rime installation 
info.
W0925 19:51:32.556815  3192 config.cc:407] nonexistent config file 
'F:\瀹氭湡鏇存柊\杞欢璁剧疆\Rime\installation.yaml'.
I0925 19:51:32.587817  3192 deployment_tasks.cc:63] creating installation info.
I0925 19:51:32.589818  3192 deployment_tasks.cc:70] generated installation id: 
4b699340-ad3a-4a98-b73c-28d0a0656e50
I0925 19:51:32.589818  3192 config.cc:316] write: installation_id
I0925 19:51:32.589818  3192 config.cc:316] write: install_time
I0925 19:51:32.589818  3192 config.cc:316] write: distribution_name
I0925 19:51:32.590817  3192 deployment_tasks.cc:81] distribution: 灏忕嫾姣?
I0925 19:51:32.590817  3192 config.cc:316] write: distribution_code_name
I0925 19:51:32.590817  3192 deployment_tasks.cc:86] distribution code name: 
Weasel
I0925 19:51:32.590817  3192 config.cc:316] write: distribution_version
I0925 19:51:32.590817  3192 deployment_tasks.cc:92] distribution version: 
0.9.15.0
I0925 19:51:32.590817  3192 config.cc:316] write: rime_version
I0925 19:51:32.590817  3192 deployment_tasks.cc:95] Rime version: 0.9.3
W0925 19:51:32.594818  3192 config.cc:407] nonexistent config file 
'F:\瀹氭湡鏇存柊\杞欢璁剧疆\Rime\default.yaml'.
I0925 19:51:32.594818  3192 config.cc:410] loading config file 'E:\Program 
Files\Rime\weasel-0.9.15\data\default.yaml'.
I0925 19:51:32.620820  3192 customizer.cc:103] updating config file 
'F:\瀹氭湡鏇存柊\杞欢璁剧疆\Rime\default.yaml'.
E0925 19:51:32.621819  3192 customizer.cc:112] Error copying config file 
'E:\Program Files\Rime\weasel-0.9.15\data\default.yaml' to user directory.
W0925 19:51:32.624819  3192 config.cc:407] nonexistent config file 
'F:\瀹氭湡鏇存柊\杞欢璁剧疆\Rime\weasel.yaml'.
W0925 19:51:36.717025  3192 config.cc:407] nonexistent config file 
'F:\瀹氭湡鏇存柊\杞欢璁



是不是因为乱码？文件路径有中文的原因呢？
```

Original comment by `shensh...@gmail.com` on 25 Sep 2012 at 11:53

**Comment by GoogleCodeExporter** [comment]
```
你好，这是三个Log文件，貌似两外两个里面路径名没有乱码��
�
麻烦解答，谢谢~~
```

Original comment by `shensh...@gmail.com` on 25 Sep 2012 at 11:59

Attachments:
- [rime.weasel.WIN-8PDHNA3BDRA.Administrator.log.WARNING.20120925-195448.2384](https://storage.googleapis.com/google-code-attachments/rimeime/issue-270/comment-3/rime.weasel.WIN-8PDHNA3BDRA.Administrator.log.WARNING.20120925-195448.2384)
- [rime.weasel.WIN-8PDHNA3BDRA.Administrator.log.INFO.20120925-195448.2384](https://storage.googleapis.com/google-code-attachments/rimeime/issue-270/comment-3/rime.weasel.WIN-8PDHNA3BDRA.Administrator.log.INFO.20120925-195448.2384)
- [rime.weasel.WIN-8PDHNA3BDRA.Administrator.log.ERROR.20120925-195448.2384](https://storage.googleapis.com/google-code-attachments/rimeime/issue-270/comment-3/rime.weasel.WIN-8PDHNA3BDRA.Administrator.log.ERROR.20120925-195448.2384)

**Comment by GoogleCodeExporter** [comment]
```
問題應該與中文的編碼有關。出於通用性的考慮RIME內部處理�
��符串統一使用UTF-8編碼。而Windows系統不同語言的版本使用不
同的字符編碼。本次增加這個功能，保存的用戶設置爲 UTF-8 
字符串，這不好用。
默認情況下，輸入法直接調用系統API獲得用戶目錄的位置，��
�以即使 %APPDATA%\Rime 
展開後包含中文，所得的路徑正好是打開文件時系統所需的GB
K/BIG5編碼。（查看以UTF-8編碼的日誌文件，會發現路徑中包含
亂碼，雖然對於Rime來說這是錯的，但文件路徑最終是爲系統�
��用）
```

Original comment by `chen....@gmail.com` on 25 Sep 2012 at 2:56
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
Original comment by `chen....@gmail.com` on 25 Sep 2012 at 3:09
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
已经解决，确实应该是路径的问题。
现在已经完美使用了，非常感谢
```

Original comment by `shensh...@gmail.com` on 25 Sep 2012 at 3:20

**Comment by GoogleCodeExporter** [comment]
```
Fixed in Weasel 0.9.15.1
```

Original comment by `chen....@gmail.com` on 26 Sep 2012 at 2:57
- Changed state: **Fixed**
